src/main/java/bio/terra/service/filedata/flight/ingest/IngestPopulateFileStateFromFileStep.java

```diff
@@ -51,26 +60,40 @@ void readFile(BufferedReader reader, FlightContext context) throws IOException {
         new ErrorCollector(
             maxBadLoadFileLineErrorsReported,
             "Invalid lines in the control file. [All lines in control file must be valid in order to proceed - 'maxFailedFileLoads' not applicable here.]");
-    long lineCount = 0;
-    List<BulkLoadFileModel> fileList = new ArrayList<>();
+    List<Future<BulkLoadFileModel>> futures = new ArrayList<>();
 
+    // Value used in a lambda so needs to be effectively final
+    final AtomicLong lineCount = new AtomicLong(0);
```

Comment:
Instead of doing this, you could introduce a local `final` next to `lineCopy` below. It's not a big deal but using `AtomicLong` is more overhead than using a `long` directly.

```
      final String lineCopy = line;
      final long lineCountCopy = ++lineCount;
```